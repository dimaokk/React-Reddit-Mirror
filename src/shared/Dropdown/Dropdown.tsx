import React, { useEffect, useState } from "react";
import styles from "./Dropdown.css";

interface IDropdownProp {
    button: React.ReactNode;
    children: React.ReactNode;
    isOpen?: boolean;
    onOpen?: () => void;
    onClose?: () => void;
}

const NOOP = () => {}

export default function Dropdown({
    button,
    children,
    isOpen,
    onClose = NOOP,
    onOpen =  NOOP,
}: IDropdownProp) {
    const [isOpenDrop, setIsOpenDrop] = useState(isOpen);

    useEffect(() => {
        setIsOpenDrop(isOpen);
    }, [isOpen]);

    useEffect(()=> {
        isOpenDrop ? onOpen() : onClose()
    }, [isOpenDrop])

    const handlrOpen = () => {
        if (isOpen === undefined) {
            setIsOpenDrop(isOpenDrop);
        }
    };
    return (
        <div className={styles.container}>
            <div onClick={() => setIsOpenDrop(!isOpenDrop)}>{button}</div>
            {isOpenDrop && (
                <div className={styles.listContainer}>
                    <div
                        className={styles.list}
                        onClick={() => setIsOpenDrop(false)}
                    >
                        {children}
                    </div>
                </div>
            )}
        </div>
    );
}
