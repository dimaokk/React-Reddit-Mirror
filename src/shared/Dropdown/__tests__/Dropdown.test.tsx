import Dropdown from "../Dropdown";
import React from "react";
import { shallow } from "enzyme";
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

describe("Dropdown", () => {
    test("shood render", () => {
        const wrapper = shallow(
            <Dropdown children={<div />} button={<button />} />
        );
        expect(wrapper).toBeDefined();
        expect(wrapper.find("button")).toBeDefined();
    });
    test('snap', () => {
        const wrapper = shallow(
            <Dropdown children={<div />} button={<button />} />
        );
        expect(wrapper).toMatchSnapshot()
    })
});

Enzyme.configure({ adapter: new Adapter() });
