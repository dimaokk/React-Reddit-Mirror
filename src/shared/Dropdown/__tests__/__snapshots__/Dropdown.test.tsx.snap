// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown snap 1`] = `
<div
  className="container"
>
  <div
    onClick={[Function]}
  >
    <button />
  </div>
</div>
`;
